aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/adapter/SQLPredicate.java

```diff
@@ -89,7 +89,7 @@ public String toString() {
      * Returns the selection arguments for the converted query string.
      * @return the selection arguments for the converted query string.
      */
-    public List<String> getSelectionArgs() {
+    public List<Object> getSelectionArgs() {
```

Comment:
Is it the duty of this `SQLPredicate`'s to prepare SQL statements? If so, SQLite will need them as `String`, right? Should we not just return `String` (or `List<String>`, here)?

The generic `Object` thing comes from the specification of predicates in the `.model.` sub-package right? But, whatever component is responsible for interpreting that into SQL, that component should stop using generic `Object` and convert to a form that a SQL database can act on.

Reply:
so the SQLite library has this thing, where assigning variable values to compiled update/insert/delete statement requires binding each value along with its specific type. That's why I changed it to object so that it can remember the data type to be converted to equivalent SQLite datatype. 

Meanwhile, for query, the selection arguments must be provided as `String[]`, which is why I initially had it as List<String> (since I implemented query before this).